components/sup/src/manager/mod.rs

```diff
@@ -669,7 +669,16 @@ impl Manager {
             Some(ref evg) => Some(events::EventsMgr::start(evg.clone())),
             None => None,
         };
-        signals::init();
+        // On Windows initializng the signal handler will create a ctrl+c handler for the
+        // process which will disable default windows ctrl+c behavior and allow us to
+        // handle via check_for_signal. However, if the supervsor is in a long running
+        // non-run hook, the below loop will not get to check_for_signal in a reasonable
+        // amount of time and the supervisor will not respond to ctrl+c. On Windows, we
+        // let the launcher catch ctrl+c and gracefully shut down services. ctrl+c should
+        // simply halt the supervisor
+        if env::var("HAB_SUP_IGNORE_SIGNAL").is_err() {
+            signals::init();
+        }
```

Comment:
I'd have a small preference for setting a constant over duplicating code. Something like
```rust
let handle_signals = env::var("HAB_SUP_IGNORE_SIGNAL").is_err()
```
But really, it's probably best to implement this as a feature and do:
```rust
if feat::is_enabled(feat::HandleSignals)
```
Either way, not a big deal, just a suggestion. I think this code should work.